src/components/Button/PButton.vue

```diff
@@ -99,35 +100,36 @@
   relative
   inline-flex
   items-center
+  bg-transparent
+  border
+  border-input
+  text-primary-foreground
   px-4
   py-2
   rounded-default
-  border
   outline-none
   focus:ring-spacing-focus-ring
   focus:ring-focus-ring
   focus:ring-offset-focus-ring
   focus:ring-offset-focus-ring-offset
-  bg-transparent
-  border-input
-  text-primary-foreground;
-}
-
-.p-button:focus:not(:focus-visible) { @apply
-  ring-transparent
-  ring-offset-transparent
+  [&:focus:not(:focus-visible)]:ring-transparent
+  [&:focus:not(:focus-visible)]:ring-offset-transparent
+  disabled:cursor-not-allowed
+  disabled:opacity-50
+  enabled:hover:border-[color:var(--p-color-text-subdued)]
+  enabled:active:bg-[color:var(--p-color-button-default-bg-active)]
+  enabled:active:border-[color:var(--p-color-text-subdued)]
+  data-[loading='true']:cursor-wait
 }
 
-.p-button:not(:disabled):hover {
-  background-color: var(--p-color-button-default-bg-hover);
-  border-color: var(--p-color-button-default-border-hover);
-  color: var(--p-color-button-default-text-hover);
-}
-
-.p-button:not(:disabled):active {
-  background-color: var(--p-color-button-default-bg-active);
-  border-color: var(--p-color-button-default-border-active);
-  color: var(--p-color-button-default-text-active);
+.p-button__content { @apply
+  gap-1
+  inline-flex
+  flex-grow
+  justify-center
+  items-center
+  font-normal
+  group-data-[loading='true']:opacity-40
```

Comment:
is this like a `has` selector? Also, is `group` something special in tailwind?

Reply:
I see, it generates `.group[data-loading=true] .p-button__content`. Is `group` a special name in tailwind? Or can it be any class? Struggling to find good docs on that. 

My concern here is if we use `group` on say `p-form` and also add a loading state using this same technique. I believe all the buttons within the form would start spinning. If the we used a modifier class like `.p-button__content--loading` or if we can use a custom class other than `group` (like `p-button-data[loading='true']:opacity-40`) 

If I'm not understanding this right please help lol.